code/goonstation/modules/power/power_transmission_laser/transmission_laser.dm

```diff
@@ -0,0 +1,448 @@
+	name = "power transmission laser"
+	desc = "Sends power over a giant laser beam to an NT power processing facility."
+
+	icon = 'icons/goonstation/objects/pt_laser.dmi'
+	icon_state = "ptl"
+
+	max_integrity = 10000
+
+	density = TRUE
+	anchored = TRUE
+
+	pixel_x = 0
+	pixel_y = -64
+
+	/// Variables go below here
+	/// The range we have this basically determines how far the beam goes its redone on creation so its set to a small number here
+	var/range = 5
+	/// Amount of power we are outputting
+	var/output_level = 0
+	/// The total capacity of the laser
+	var/capacity = INFINITY
+	/// Our current charge
+	var/charge = 0
+	/// Should we try to input charge paired with the var below to check if its fully inputing
+	var/input_attempt = TRUE
+	/// Are we currently inputting
+	var/inputting = TRUE
+	/// The amount of charge coming in from the inputs last tick
+	var/input_available = 0
+	/// Have we been switched on?
+	var/turned_on = FALSE
+	/// Are we attempting to fire the laser currently?
+	var/firing = FALSE
+	/// We need to create a list of all lasers we are creating so we can delete them in the end
+	var/list/laser_effects = list()
+	/// List of all blocking turfs or objects
+	var/list/blocked_objects = list()
+	/// Our max load we can set
+	var/max_grid_load = 0
+	/// Our current grid load
+	var/current_grid_load = 0
+	/// Out power formatting multiplier used inside tgui to convert to things like mW gW to watts for ease of setting
+	var/power_format_multi = 1
+	/// Same as above but for output
+	var/power_format_multi_output = 1
+
+	/// Are we selling the energy or just sending it into the ether
+	var/selling_energy = FALSE
+
+	/// How much energy have we sold in total (Joules)
+	var/total_energy = 0
+	/// How much energy do you have to sell in order to get an announcement
+	var/announcement_threshold = 1 MJ
+
+	/// How much credits we have earned in total
+	var/total_earnings = 0
+	/// The amount of money we haven't sent to cargo yet
+	var/unsent_earnings = 0
+
+	/// How much we are inputing pre multiplier
+	var/input_number = 0
+	/// How much we are outputting pre multiplier
+	var/output_number = 0
+	/// Our set input pulling
+	var/input_pulling = 0
+	/// Announcement configuration for updates
+	var/datum/announcer/announcer = new(config_type = /datum/announcement_configuration/ptl)
+	/// Last direction the laser was pointing. So offset doesn't get handles when it doesn't need to
+	var/last_dir = 0
+
+
+/obj/machinery/power/transmission_laser/Initialize(mapload)
+	. = ..()
+	range = get_dist(get_step(get_front_turf(), dir), get_edge_target_turf(get_front_turf(), dir))
+	if(!powernet)
+		connect_to_network()
+	handle_offset()
+	update_icon()
+
+/obj/machinery/power/transmission_laser/proc/handle_offset()
+	switch(dir)
+		if(NORTH)
+			pixel_x = -64
+			pixel_y = 0
+			var/datum/component/oldmap = GetComponent(/datum/component/multitile)
+			if(oldmap)
+				qdel(oldmap)
+			AddComponent(/datum/component/multitile, list(
```

Comment:
The component should handle this when adding a new one. If it currently doesn't, then you'll want to update its `dupe_type` to force this behavior.

Reply:
done and tested. it works fine without it.